src/PrestaShopBundle/Translation/Provider/AbstractProvider.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * 2007-2016 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ *  @author 	PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2016 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+
+class AbstractProvider implements ProviderInterface
+{
+    use TranslationFinderTrait;
+
+    private $databaseLoader;
+    private $resourceDirectory;
+    private $locale;
+
+    public function __construct(LoaderInterface $databaseLoader, $resourceDirectory)
+    {
+        $this->databaseLoader = $databaseLoader;
+        $this->resourceDirectory = $resourceDirectory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDirectories()
+    {
+        return array('');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTranslationDomains()
+    {
+        return array('');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLocale()
+    {
+        return $this->locale;
+    }
+
+    public function setLocale($locale)
+    {
+        $this->locale = $locale;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMessageCatalogue()
+    {
+        return new MessageCatalogue();
+    }
+
+    public function getResourceDirectory()
+    {
+        return $this->resourceDirectory.'/'.$this->locale.'/';
```

Comment:
Please just return `return $this->resourceDirectory.'/'.$this->locale`

All directory methods of php return the path without the last directory separator. If we add some methods return with and some without. We have to maintain only one return type for paths.

And yep: many old methods of PrestaShop return the paths with the last directory separator, but we have to stop it.